contrib/win32/win32compat/ssh-agent/keyagent-request.c

```diff
@@ -245,8 +339,11 @@ static int sign_blob(const struct sshkey *pubkey, u_char ** sig, size_t *siglen,
 		free(regdata);
 	if (tmpbuf)
 		sshbuf_free(tmpbuf);
-	if (prikey)
-		sshkey_free(prikey);
+#ifdef ENABLE_PKCS11
+	if (!is_pkcs11_key)
+#endif /* ENABLE_PKCS11 */
+		if (prikey)
+			sshkey_free(prikey);
```

Comment:
It is safe to call `sshkey_free(NULL)`, which would simplify this code a bit.

Reply:
If the key is a pkcs11 key (`is_pkcs11_key = 1`) then `prikey` will be a pointer from `pkcs11_keylist` and we don't want to free it here (rather, we'll free it in the caller - `process_sign_request()` as the allocation happens at the same level).
In other words, we only want to free `prikey` if it is associated with a non-pkcs11 key here.
We could remove the line `if (prikey)` but I would rather follow the convention at the cost of an additional line.
Let me know if I misunderstood your suggestion here.